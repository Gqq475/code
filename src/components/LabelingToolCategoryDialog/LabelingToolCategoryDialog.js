import React from 'react'
import classes from './LabelingToolCategoryDialog.scss'
import { Button, Dialog } from '@blueprintjs/core'
import Tabs from 'components/LabelingToolDefectType'
import OurToaster from 'components/OurToaster'
import { connect } from 'react-redux'
const qca = __QCA__

type Props = {
  data: Object,
  isOpen: boolean,
  toggleDialog: Function,
  addDefectType: Function,
  getDefectTypeList: Function,
  token: String
};

export class LabelingToolCategoryDialog extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleDelete = this.handleDelete.bind(this)
    this.state = {
      factoryCode: 'TFT',
      departmentCode: 'General',
      name: ''
    }
  }

  handleChange (type) {
    return (e) => {
      this.setState({ [type]: e.target.value })
    }
  }

  handleSubmit () {
    this.props.addDefectType(this.state)
    this.props.toggleDialog()
    var self = this
    setTimeout(() => { self.props.getDefectTypeList() }, 1000)
  }

  handleDelete () {
    const url = `${qca}/offline/labeling/defect/type`
    if (this.props.data.currentDefectType) {
      fetch(url, {
        method: 'DELETE',
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
          'Authorization': this.props.token
        },
        body: JSON.stringify({
          id: this.props.data.currentDefectType.defectTypeId
        })
      })
      .then(response => response.json())
      .then(data => {
        if (data.code === 200000) {
          OurToaster.show({message: '删除成功！'})
        } else {
          OurToaster.show({message: '删除失败！'})
        }
      })
      this.props.toggleDialog()
      var self = this
      setTimeout(() => { self.props.getDefectTypeList() }, 1000)
    } else {
      OurToaster.show({message: '请选择删除的类别!'})
    }
  }

  render () {
    return (
      <div className={classes['LabelingToolCategoryDialog-container']}>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={classes.headerTitle}>类别修改</h6>
          </div>
          <div className={classes.dialogBody}>
            <div className={classes.leftBox}>
              <Tabs
                isDialog
                data={this.props.data}
                getDefectTypeList={this.props.getDefectTypeList} />
              <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleDelete}>
               Delete
              </Button>
            </div>
            <div className={classes.rightBox}>
              <label className={`pt-label pt-inline ${classes.filter}`}>
                厂别
                <div className='pt-select'>
                  <select className={classes.select} onChange={this.handleChange('factoryCode')}>
                    <option value='TFT'>TFT</option>
                    <option value='CF'>CF</option>
                    <option value='Cell'>Cell</option>
                    <option value='Module'>Module</option>
                  </select>
                </div>
              </label>
              <label className={`pt-label pt-inline ${classes.filter}`}>
                部门
                <div className='pt-select'>
                  <select className={classes.select} onChange={this.handleChange('departmentCode')}>
                    <option value='General'>General</option>
                    <option value='Photo'>Photo</option>
                    <option value='ThinFilm'>Thin Film</option>
                    <option value='Etch'>Etch</option>
                    <option value='Testing'>Testing</option>
                    <option value='CF'>CF</option>
                  </select>
                </div>
              </label>
              <label className={`pt-label pt-inline ${classes.filter}`}>
                类别
                <input type='text' className='pt-select' onChange={this.handleChange('name')} />
              </label>
            </div>
            <div className={classes.btnGroup}>
              <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
               Add
              </Button>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                {_.CANCEL}
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}
export default connect(
  (state) => ({
    token: state.auth.token
  }),
  (dispatch, state) => {
    return {
      dispatch
    }
  }
)(LabelingToolCategoryDialog)
