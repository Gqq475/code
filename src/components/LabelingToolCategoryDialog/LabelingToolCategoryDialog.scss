.LabelingToolCategoryDialog-container {

}
.dialog {
    width:650px;
    padding:2px;
    background: #F7F7F7;
  }
  .leftBox {
    float: left;
    width: 60%;
  }
  .rightBox {
    float:right;
    width: 38%;
    margin-left: 2%;
    margin-top: 20px;
  }
   .headerTitle {
    margin: 10px 0px 10px 15px;
    font-size: 16px;
  }

  .headerID {
    margin-left: 15px;
    font-size: 18px;
    padding-bottom: 16px;
  }
  .dialogBody {
    background: #fff;
    color: #000;
  }
  .filter {
    display: flex;
    align-items: center;
    justify-content: space-between;
    height:45px;
    line-height:45px;
    margin:0px 10px;
    font-size: 14px;
  }
  .select {
    cursor: pointer;
    outline: 0;
    border: 0;
    background: inherit;
    box-shadow: none;
    margin: 0;
    border: 1px solid #D9D9D9;
    height: 30px;
    width: 160px;
    border-radius: 3px;
  }
  .btnGroup {
    margin:17px 0px;
    clear: right;
    float: right;
  }
  .btnEdit {
    background: #0087DC;
    color: #fff;
    margin: 0px 10px 0px 15px;
    width: 80px;
    height: 35px;
    font-size: 14px;
  }
  .btnCancel {
    background: #fff;
    border: 1px solid #0087DC !important;
    color: #0087DC !important;
    width: 80px;
    height: 35px;
    font-size: 14px;
  }
