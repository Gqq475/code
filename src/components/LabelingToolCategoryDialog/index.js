import LabelingToolCategoryDialog from './LabelingToolCategoryDialog'
export default LabelingToolCategoryDialog
