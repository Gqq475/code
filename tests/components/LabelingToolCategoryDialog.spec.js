import React from 'react'
import { LabelingToolCategoryDialog } from 'components/LabelingToolCategoryDialog/LabelingToolCategoryDialog'

describe('(Component) LabelingToolCategoryDialog', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <LabelingToolCategoryDialog />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
