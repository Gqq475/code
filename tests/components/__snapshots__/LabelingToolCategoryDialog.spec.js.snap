// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) LabelingToolCategoryDialog render normally 1`] = `
<div
  className="LabelingToolCategoryDialog-container"
>
  <Dialog
    className="dialog"
    isOpen={false}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        类别修改
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <div
        className="leftBox"
      >
        <LabelingToolDefectType
          isDialog={true}
        />
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Delete
        </Blueprint.Button>
      </div>
      <div
        className="rightBox"
      >
        <label
          className="pt-label pt-inline filter"
        >
          厂别
          <div
            className="pt-select"
          >
            <select
              className="select"
              onChange={[Function]}
            >
              <option
                value="TFT"
              >
                TFT
              </option>
              <option
                value="CF"
              >
                CF
              </option>
              <option
                value="Cell"
              >
                Cell
              </option>
              <option
                value="Module"
              >
                Module
              </option>
            </select>
          </div>
        </label>
        <label
          className="pt-label pt-inline filter"
        >
          部门
          <div
            className="pt-select"
          >
            <select
              className="select"
              onChange={[Function]}
            >
              <option
                value="General"
              >
                General
              </option>
              <option
                value="Photo"
              >
                Photo
              </option>
              <option
                value="ThinFilm"
              >
                Thin Film
              </option>
              <option
                value="Etch"
              >
                Etch
              </option>
              <option
                value="Testing"
              >
                Testing
              </option>
              <option
                value="CF"
              >
                CF
              </option>
            </select>
          </div>
        </label>
        <label
          className="pt-label pt-inline filter"
        >
          类别
          <input
            className="pt-select"
            onChange={[Function]}
            type="text"
          />
        </label>
      </div>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Add
        </Blueprint.Button>
        <Blueprint.Button
          className="btnCancel"
        />
      </div>
    </div>
  </Dialog>
</div>
`;
